### PR TITLE
Storecontinuations

### PR DESCRIPTION
**Pull request by tballmsft**
I've generated, for each function, an array of the continuations in that function (JS only). I've also opened up the runtime to allow the simulator code in a target to overwrite the continuation. This is needed for my thumb simulator.

### CODE REVIEW

File: pxtlib/emitter/backjs.ts [hunk 1]
```diff
@@ -127,7 +128,7 @@ switch (step) {
         let info = nodeLocationInfo(proc.action) as FunctionLocationInfo
         info.functionName = proc.getName()
         writeRaw(`${proc.label()}.info = ${JSON.stringify(info)}`)
-
+        writeRaw(`${proc.label()}.continuations = [ ${async_continuations.join(",")} ]`)
```
Comment (pelikhan) [inline pxtlib/emitter/backjs.ts]: do we need to write this if ``async_continuations`` is empty? (note: js typeically uses camelized variable names)

File: pxtsim/runtime.ts [hunk 2]
```diff
@@ -222,13 +222,16 @@ namespace pxsim {
...
+
         dead = false;
         running = false;
         startTime = 0;
         id: string;
         globals: any = {};
         currFrame: StackFrame;
+        entry: LabelFn;
 
+        overwriteResume: (retPC: number) => void;
```
Comment (pelikhan) [inline pxtsim/runtime.ts]: is this used anywhere?

Comment (tballmsft) [inline pxtsim/runtime.ts]: yes, it is used in my target!

File: pxtlib/emitter/backjs.ts (hunk 1 above)
Comment (tballmsft) [inline pxtlib/emitter/backjs.ts]: yes, but I only need to generate the array for the main function. I will revise

Comment (tballmsft) [inline pxtlib/emitter/backjs.ts]: I tested it. No issues. pxt-microbit does not use my changes.

**Pull request merged**